### PR TITLE
stm32h7: use DMA over with uart serial peripheral

### LINKED ISSUE #37701: stm32:  conflicts with uart serial DMA
**Describe the bug**
When  `CONFIG_UART_ASYNC_API=y` throws error
It was working fine on ` VERSION_MAJOR = 2VERSION_MINOR = 6PATCHLEVEL = 0VERSION_TWEAK = 0EXTRAVERSION = rc1`

additional info (relevant) ones of how my DT looks like:
```
&usart3 {
        current-speed = <9600>; //5000K baudrate
    	dmas =  <&dmamux1 1 46 0x440>, 
		<&dmamux1 2 45 0x480>;
	dma-names = "tx", "rx";	  
        status = "okay";
};
```

I noticed this issue after moving to latest mainline code

**To Reproduce**
1. Build the code

**Expected behavior**
Build successfully 

**Impact**
Could not build the code. 

**Logs and console output**
```
t_stm32.c.obj -c /home/manoj/zephyrproject/zephyr/drivers/serial/uart_stm32.c
In file included from /home/manoj/zephyrproject/zephyr/include/sys/util_macro.h:34,
                 from /home/manoj/zephyrproject/zephyr/include/sys/util.h:17,
                 from /home/manoj/zephyrproject/zephyr/include/kernel/sched_priq.h:9,
                 from /home/manoj/zephyrproject/zephyr/include/kernel_includes.h:23,
                 from /home/manoj/zephyrproject/zephyr/include/kernel.h:17,
                 from /home/manoj/zephyrproject/zephyr/drivers/serial/uart_stm32.c:17:
zephyr/include/generated/devicetree_unfixed.h:9613:36: error: 'DT_N_S_soc_S_serial_40004800_P_dmas_NAME_rx_VAL_features' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_serial_40004800_P_dmas_NAME_rx_VAL_channel'?
 9613 | #define DT_N_INST_1_st_stm32_uart  DT_N_S_soc_S_serial_40004800
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/manoj/zephyrproject/zephyr/include/sys/util_internal.h:67:26: note: in definition of macro '__DEBRACKET'
   67 | #define __DEBRACKET(...) __VA_ARGS__
```

### PR DESCRIPTION
**Pull request by FRASTM**
With this changes, the stm32h7 is configuring the DMA (type V1) for using over a peripheral like the UART with a DMAMUX.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/37701

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
With this patch and when PD5 and PD6 for usart2 are connected on the nucleo_h723zg board (pin 6 = pin 4 of the CN9) the `tests/drivers/uart/uart_async` PASSED:

```
** Booting Zephyr OS build zephyr-v2.6.0-2019-gff7f699219d0  ***
Running test suite uart_async_test
===================================================================
START - test_single_read_setup
 PASS - test_single_read_setup in 0.1 seconds
===================================================================
START - test_single_read
 PASS - test_single_read in 0.402 seconds
===================================================================
START - test_chained_read_setup
 PASS - test_chained_read_setup in 0.1 seconds
===================================================================
START - test_chained_read
 PASS - test_chained_read in 0.267 seconds
===================================================================
START - test_double_buffer_setup
 PASS - test_double_buffer_setup in 0.1 seconds
===================================================================
START - test_double_buffer
 PASS - test_double_buffer in 3.397 seconds
===================================================================
START - test_read_abort_setup
 PASS - test_read_abort_setup in 0.1 seconds
===================================================================
START - test_read_abort
 PASS - test_read_abort in 1.110 seconds
===================================================================
START - test_chained_write_setup
 PASS - test_chained_write_setup in 0.1 seconds
===================================================================
START - test_chained_write
 PASS - test_chained_write in 0.2 seconds
===================================================================
START - test_long_buffers_setup
 PASS - test_long_buffers_setup in 0.1 seconds
===================================================================
START - test_long_buffers
 PASS - test_long_buffers in 0.151 seconds
===================================================================
START - test_write_abort_setup
 PASS - test_write_abort_setup in 0.1 seconds
===================================================================
START - test_write_abort
 PASS - test_write_abort in 0.102 seconds
===================================================================
START - test_forever_timeout_setup
 PASS - test_forever_timeout_setup in 0.1 seconds
===================================================================
START - test_forever_timeout
 PASS - test_forever_timeout in 3.1 seconds
===================================================================
Test suite uart_async_test succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

File: boards/arm/nucleo_h723zg/nucleo_h723zg.dts [hunk 1]
```diff
@@ -95,6 +95,11 @@
 	status = "okay";
 };
 
+&usart2 {
+	pinctrl-0 = <&usart2_tx_pd5 &usart2_rx_pd6>;
+	current-speed = <115200>;
+};
```
Comment (ABOSTM) [inline boards/arm/nucleo_h723zg/nucleo_h723zg.dts:101]: Missing:
`status = "okay";`

Comment (FRASTM) [reply boards/arm/nucleo_h723zg/nucleo_h723zg.dts:101]: It is enabled in the testcase (overlay), not to impact other applications.

Comment (FRASTM) [reply boards/arm/nucleo_h723zg/nucleo_h723zg.dts:102]: changed, now enabling in the nulceo_h723zg board

**Comment by FRASTM** [comment]
rebase

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by dcpleung — APPROVED** [review]

**Comment by ycsin** [comment]
I wonder if the uart serial DMA here can be used for serial peripheral like a GSM modem?

**Pull request merged**